pkg/generate/app/cmd/newapp.go

```diff
@@ -919,7 +973,21 @@ func (c *AppConfig) run(acceptors app.Acceptors) (*AppResult, error) {
 		return nil, err
 	}
 
-	if err := c.resolve(components); err != nil {
+	var imageComp app.ComponentReference
+	imageComp, repositories, err = c.addImageSource(repositories)
```

Comment:
(asking/commenting to try to understand better new-app's code base...)

When I read "validate" I think of a method that should have no side effects and return validation errors.
I was thinking why should calling `c.validate()` also "add Image Source"?! Then looking at the godoc for validate: "validate converts all of the arguments on the config into references to objects, or returns an error".

In my head addImageSource matches the godoc for `validate`, but the name "validate" makes no sense to me in this context, is it just me?!